lib/charms/acme_client_operator/v0/acme_client.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""# acme_client Library.
+
+This library is designed to enable developers to easily create new charms for the ACME protocol.
+This library contains all the logic necessary to get certificates from an ACME server..
+
+## Getting Started
+To get started using the library, you need to fetch the library using `charmcraft`.
+```shell
+charmcraft fetch-lib charms.acme_client_operator.v0.acme_client
+```
+You will also need to add the following library to the charm's `requirements.txt` file:
+- jsonschema
+- cryptography
+
+Then, to use the library in an example charm, you can do the following:
+```python
+from charms.acme_client_operator.v0.acme_client import AcmeClient
+from ops.main import main
+class ExampleAcmeCharm(AcmeClient):
+    def __init__(self, *args):
+        super().__init__(*args, plugin="namecheap")
+        self._server = "https://acme-staging-v02.api.letsencrypt.org/directory"
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
+    def _on_config_changed(self, _):
+        try:
+            self.update_generic_acme_config(
```

Comment:
This method doesn't exist in the base AcmeClient.

Reply:
✅

Reply:
Wait... Did you change anything about that? I can't see any changes ;)

Reply:
oh, didn't push everything sorry for that.
Now it's supposed to be visible :)